sentinelhub/sentinelhub_request.py

```diff
@@ -2,31 +2,22 @@
 
 Documentation: https://docs.sentinel-hub.com/api/latest/reference/
 """
+import warnings
 
-from .constants import MimeType, DataSource, RequestType
+from .constants import MimeType, RequestType
+from .data_collections import DataCollection, OrbitDirection
 from .download import DownloadRequest, SentinelHubDownloadClient
 from .data_request import DataRequest
+from .exceptions import SHDeprecationWarning
 from .geometry import Geometry, BBox
 from .time_utils import parse_time_interval
 
 
-def _update_other_args(dict1, dict2):
-    """
-    Function for a recursive update of `dict1` with `dict2`. The function loops over the keys in `dict2` and
-    only the non-dict like values are assigned to the specified keys.
-    """
-    for key, value in dict2.items():
-        if isinstance(value, dict) and key in dict1:
-            _update_other_args(dict1[key], value)
-        else:
-            dict1[key] = value
-
-
 class SentinelHubRequest(DataRequest):
     """ Sentinel Hub API request class
     """
     def __init__(self, evalscript, input_data, responses, bbox=None, geometry=None, size=None, resolution=None,
-                 **kwargs):
+                 base_url=None, **kwargs):
```

Comment:
Wouldn't it be better that base_url is taken from DataCollection upon requesting data?

Or is there any other use-case, where duplicity is needed?